compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3689,10 +3689,19 @@ public BVarSymbol defineVarSymbol(Location pos, Set<Flag> flagSet, BType varType
         // Create variable symbol
         Scope enclScope = env.scope;
         BVarSymbol varSymbol = createVarSymbol(flagSet, varType, varName, env, pos, isInternal);
-
-        // Add it to the enclosing scope
-        if (!symResolver.checkForUniqueSymbol(pos, env, varSymbol)) {
-            varSymbol.type = symTable.semanticError;
+        if (flagSet.contains(Flag.FIELD)) {
+            if (!symResolver.checkForUniqueMemberSymbol(pos, env, varSymbol)) {
+                varSymbol.type = symTable.semanticError;
+            }
+        } else if (flagSet.contains(Flag.REQUIRED_PARAM) || flagSet.contains(Flag.DEFAULTABLE_PARAM) ||
+                flagSet.contains(Flag.REST_PARAM)) {
+            if (!symResolver.checkForUniqueMemberSymbol (pos, env, varSymbol)) {
+                varSymbol.type = symTable.semanticError;
+            }
```

Comment:
Can't we combine these two? We are doing the same thing in both blocks, right?

Reply:
Shouldn't this be a generic fix anyway? For example, even with this change, the following wouldn't work, right?

```ballerina
public function main() {
    int 'error;
}
```

Reply:
IMO, it will be better if we can fix this in `org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolResolver#isSymbolDefinedInRootPkgLvl`.

I think we can even consider special casing the types such as json and error that we know will cause this conflict since this package is totally under our control.

Reply:
Are you suggesting to return false to `error`, `json` and `anydata` like symbols?

Reply:
Yeah, either that or if we can change the `(foundSym.tag & SymTag.VARIABLE_NAME) == SymTag.VARIABLE_NAME` check to return the result we want here.

`json`, `anydata`, `error` etc. here should be similar to other built-in types, right?

Reply:
Don't think all built-in types share the same issue. Many will have SymTag as `TYPE`. This starts with `TYPE_DEF` which also has `VARIABLE_NAME` flag. `ERROR` is also a `TYPE_DEF`

Reply:
Sorry, what I meant was that the behaviour or rather the contexts in which they can be used should be the same.